utils/src/main/scala/cromwell/pipeline/utils/ApplicationConfig.scala

```diff
@@ -88,9 +96,25 @@ class ApplicationConfig(val config: Config) {
     )
   }
 
+  lazy val postgreConfig: PostgreConfig = {
+    val _config = config.getConfig("database.postgres_dc.db.properties")
+    PostgreConfig(
+      serverName = _config.getString("serverName"),
+      portNumber = _config.getInt("portNumber"),
+      databaseName = _config.getString("databaseName"),
+      user = _config.getString("user"),
+      password = _config.getString("password").toCharArray
+    )
+  }
+
 }
 
 object ApplicationConfig {
   def load(config: Config = ConfigFactory.load()): ApplicationConfig =
     new ApplicationConfig(config)
+
+  def unapply(
+    config: ApplicationConfig
+  ): Option[(WebServiceConfig, AuthConfig, GitLabConfig, MongoConfig, PostgreConfig)] =
```

Comment:
I think, you can to use some trait like SomeConfig or different name for combine all configs under one type. It shod help if you want to add a new config.

Reply:
Great idea. Can you be more specific? Can we discuss it in teams?